addons/hr_holidays/report/hr_leave_employee_report.py

```diff
@@ -0,0 +1,153 @@
+            report_record['is_hatched'] = report_record['state'] not in ['refuse', 'validate']
+
+    def _create_report_records_from_leave_records(self, leave_records, leave_field_names):
+        report_records = [
+            {field_name: leave_record[index] for index, field_name in enumerate(leave_field_names)}
+            for leave_record in leave_records
+        ]
+        return report_records
+
+    def _compute_working_schedule_aligned_dates(self, report_records):
+        for report_record in report_records:
+            start_date = report_record['day_aligned_date_from'].replace(tzinfo=utc)
+            end_date = report_record['day_aligned_date_to'].replace(tzinfo=utc)
+            work_intervals = list(self.env['hr.leave'].browse(report_record['leave_id'])
+                .resource_calendar_id._work_intervals_batch(start_date, end_date, compute_leaves=False)[False])
+            if work_intervals:
+                report_record['working_schedule_aligned_date_from'] = work_intervals[0][0].astimezone(utc).replace(tzinfo=None)
+                report_record['working_schedule_aligned_date_to'] = work_intervals[-1][1].astimezone(utc).replace(tzinfo=None)
+            else:
+                # Fallback if no work intervals found
+                report_record['working_schedule_aligned_date_from'] = start_date.replace(tzinfo=None)
+                report_record['working_schedule_aligned_date_to'] = end_date.replace(tzinfo=None)
+
+    def _compute_leave_duration(self, report_records):
+        leave_ids = [report_record['leave_id'] for report_record in report_records]
+        leaves = self.env['hr.leave'].browse(leave_ids)
+        holiday_status_id_by_leave_id = {leave.id: leave.holiday_status_id.id for leave in leaves}
+        for report_record in report_records:
+            # Remove the day_aligned_date_from/to as they are intermediate values for computing other values
+            virtual_leave = self.env['hr.leave'].new({
+                'date_from': report_record.pop('day_aligned_date_from'),
+                'date_to': report_record.pop('day_aligned_date_to'),
+                'employee_id': report_record['employee_id'],
+                'holiday_status_id': holiday_status_id_by_leave_id[report_record['leave_id']],
+            })
+            leave_duration = virtual_leave._get_durations(additional_domain=[('holiday_id', '!=', report_record['leave_id'])])[virtual_leave.id]
+            report_record['number_of_days'] = leave_duration[0]
+            report_record['number_of_hours'] = leave_duration[1]
+
+    def _generate_report_query(self, report_records):
+        report_records = [report_record for report_record in report_records if report_record.get('number_of_days', 0) > 0]
+        if not report_records:
+            return ""
+
+        column_names = list(report_records[0].keys())
+        report_records_tuples = [tuple(report_record[column_name] for column_name in column_names) for report_record in report_records]
```

Comment:
I think this line could be more readable like this or we can extract the outer loop to be normal `for`
```suggestion
        report_records_tuples = [
            tuple(report_record[column_name] for column_name in column_names) 
            for report_record in report_records
        ]
```